dentapp/src/components/loginInput.js

```diff
@@ -24,42 +29,38 @@ font-style: normal;
 font-weight: bold;
 font-size: 12px;
 line-height: 12px;
-color: ${(props) => (props.primary ? "#FFFFFF" : "#4A4A4A")};
+color: ${(props) => (props.theme ? white : darkgrey)};
 `;
 
 class LoginInput extends React.Component {
   state = {
     input: "",
-    placeholder: "contact@email.com",
-  };
-
-  handleChange = (e) => {
-    this.setState({ input: e.target.value });
-    console.log("input value: ", e.target.value);
   };
 
   render() {
     return (
       <>
-        <InputLabel primary> Mail </InputLabel>
-        <InputGroup>
-          <Form
-            primary
-            placeholder={"contact@email.com"}
-            onChange={this.handleChange}
-          />
-        </InputGroup>
-
-        <InputLabel> Mail </InputLabel>
+        <InputLabel> {props.label} </InputLabel>
         <InputGroup>
-          <Form
-            placeholder={this.state.placeholder}
-            onChange={this.handleChange}
-          />
+          <Form placeholder={props.actionText} onChange={props.handleChange} />
         </InputGroup>
       </>
     );
   }
 }
 
 export default LoginInput;
+
+Form.PropTypes = {
+  theme: PropTypes.border({
+    primary: PropTypes.string.isRequired,
+    secondary: PropTypes.string.isRequired,
+  }),
+};
+
+InputLabel.PropTypes = {
+  theme: PropTypes.color({
+    primary: PropTypes.string.isRequired,
+    secondary: PropTypes.string.isRequired,
+  }),
+};
```

Comment:
All PropTypes should be on LoginInput , there are props validation missing: actionText, actionText, handleChange